docs/jira-upgrade.md

```diff
@@ -0,0 +1,34 @@
+## Jira Rolling Upgrade
+Suppose we have a Jira cluster with 3 nodes, all running `Jira xx.xx.0-jdk11`, and an upgrade to
+`Jira xx.xx.1-jdk11` is planned. Use these steps to complete the upgrade process:
+
+1. Update the `version` and `appVersion` in the Helm chart (`src/main/charts/jira/Chart.yaml`):
+ ```yaml
+ version: 0.1.1
+ appVersion: xx.xx.1-jdk11
+ ```
+
+2. Create a new version of the Helm package:
+ ```shell script
+ $ helm package src/main/charts/jira --destination target/helm
```

Comment:
As discussed IRL we should not encourage people to package the chart from the source files. Instead, they can set the `image.tag` if they want to change the app version.

Ideally there would be a parameter for `helm upgrade` command but it is currently just an open issue - https://github.com/helm/helm/issues/8194